### PR TITLE
A note on security of IAction objects

### PR DESCRIPTION
**Pull request by dahlia**
I elaborate on the relationship of `IAction` code and its property values from a perspective of security.

Preview: <https://docs.libplanet.io/action-docs/api/Libplanet.Action.IAction.html>

### CODE REVIEW

**Comment by dahlia** [comment]
I accidentally made a branch in this upstream repository, not my own fork, so this made two sets of builds (on all GitHub Actions, Travis CI, and Azure Pipelines), and one (as a branch push) of those failed, but other one (as a pull request) succeeded.  Nevermind these red badges!

File: Libplanet/Action/IAction.cs [hunk 1]
```diff
@@ -14,6 +14,27 @@ namespace Libplanet.Action
...
+    /// That means if you define an action class it also defines what every
+    /// honest node can do in the network.  Even if a malicious node changes
+    /// their own action code it won't affect other honest nodes in
+    /// the network.</para>
+    /// <para>For example, where honest nodes share the common action
+    /// <c>Heal(Target) => PreviousStates[Target] + 1</c>, suppose a malicious
+    /// node <c>m</c> changes their own <c>Heal</c> action code to
+    /// <c>Heal(Target) => PreviousStates[Target] + 2</c> (2 instead of 1),
+    /// and then send an action <c>Heal(m)</c>.
+    /// Fortuneately, this action does not work as <c>m</c>'s intention,
```
Comment (earlbread) [inline Libplanet/Action/IAction.cs]:
```suggestion
    /// Fortunately, this action does not work as <c>m</c>'s intention,
```

File: Libplanet/Action/IAction.cs [hunk 2]
```diff
@@ -14,6 +14,27 @@ namespace Libplanet.Action
...
+    /// because the changed code in itself is not used by other honest nodes,
+    /// so they still increase only 1, not 2.  The effect of that double healing
+    /// is a sort of &#x201c;illusion&#x201d; only visible to the malicious node
+    /// alone.</para>
+    /// <para>In conclusion, action code is a part of protocol and it works with
```
Comment (earlbread) [inline Libplanet/Action/IAction.cs]:
```suggestion
    /// <para>In conclusion, action code is a part of the protocol and it works with
```

File: Libplanet/Action/IAction.cs [hunk 3]
```diff
@@ -14,6 +14,27 @@ namespace Libplanet.Action
...
+    /// consensus in the network, so only things each node can affect the network
+    /// in general is property values of each action they signs and sends,
```
Comment (earlbread) [inline Libplanet/Action/IAction.cs]:
```suggestion
    ///, in general, is property values of each action they sign and sends,
```

Comment (dahlia) [reply Libplanet/Action/IAction.cs]: I think this is not appropriate, as “in general” here is not a conjunction.

File: Libplanet/Action/IAction.cs [hunk 4]
```diff
@@ -14,6 +14,27 @@ namespace Libplanet.Action
...
+    /// <c>Heal(Target) => PreviousStates[Target] + 2</c> (2 instead of 1),
+    /// and then send an action <c>Heal(m)</c>.
+    /// Fortunately, this action does not work as <c>m</c>'s intention,
+    /// because the changed code in itself is not used by other honest nodes,
+    /// so they still increase only 1, not 2.  The effect of that double healing
+    /// is a sort of &#x201c;illusion&#x201d; only visible to the malicious node
+    /// alone.</para>
+    /// <para>In conclusion, action code is a part of the protocol and it works with
+    /// consensus in the network, so only things each node can affect the network
+    /// in general is property values of each action they signs and sends,
```
Comment (dahlia) [inline Libplanet/Action/IAction.cs]:
```suggestion
    /// in general is property values of each action they sign and send,
```

**Pull request merged**